### ISSUE TITLE
Widen definition of 'Contributors' (cont.)

### ISSUE THREAD

**Issue by Beanow** [sfosc/sfosc#62]
This discussion started as part of #24 and I think it's worth continuing outside of a pull request.

It's about the different terms in https://sfosc.org/sfosc-book/rawls_for_foss/
Particularly:

> - Developers: people with the technical ability and desire to improve the software; for their own benefit, the benefit of their employer, the desire to be active in the community or a combination of these.
> - Contributors: those who engage in the work of making the software better by not by writing code. They might file bug reports, compile release notes, write documentation, answer questions, etc.

The PR raises a few points:
- Should Contributors include code contributions, even though there's a Developers term for this?
- Is Contributors the right term to be using for community members that don't contribute code?
- Do we need more terms to cover this correctly?

Let's explore that a bit more in this issue.

**Comment by raboof** [comment]
> should Contributors include code contributions

Yes, I think this is the way the term is commonly used (even in the rest of the sfosc text), so it would make sense to define it like this

> Is Contributors the right term to be using for community members that don't contribute code?

This is perhaps less universal, but I think it is good to affirm non-code contributions are as important as (sometimes more important than) code contributions, so calling those who make non-code contributions "contributors" makes sense to me.

> Do we need more terms to cover this correctly

Not AFAICS

**Comment by Beanow** [comment]
I very much agree with @adamhjk on:
> I think we do need to have separate words. I think it's important that we recognize the different motivations for different constituencies, all of which we are trying to serve at the same time in the community. They are all contributors.

Having a more holistic view of all the different parties involved in a sustainable project to me seems essential. And as he eluded to:

> For example, the venture backers of Chef have put, collectively, a large amount of money into the community. Historically, we wouldn't have talked about that as a contribution - instead, we would say contributors are the people working on the project, but certainly not the capital source.

I think it starts with identifying and acknowledging them as best we can.

The term that complicates things is probably the Contributor one. Because it historically meant something else in this context and is *still reinforced* by how tools like GitHub define and display it.

![Contributor - This user has previously committed to the sfosc repository.](https://user-images.githubusercontent.com/497556/57445936-bf5e6f80-7253-11e9-89f7-244724210cf8.png)
_(This badge is really unfortunate, we ought to lobby with GitHub about that.)_

I don't think we have to follow this definition though, because Contributor is used in many other contexts to mean other things. Charity contributor, Wiki contributor, Community contributor, ...

So I think, more so than new terms to avoid confusion with this one. We need a common practice of acknowledging more forms of contribution besides git commits. Odds are, we're going to miss a lot of them anyway. So we could provide a live document of examples at best?

Now I noticed for example [js-ipfs](https://github.com/ipfs/js-ipfs/blob/master/package.json#L194) took the trouble of updating their contributor list in the package.json according to [it's spec](https://docs.npmjs.com/files/package.json#people-fields-author-contributors) and realized it isn't particularly helpful. It should imo include at least a rough indication of what they've done to help.

So perhaps we can reformat this to explain, these are non-exhaustive examples and you are strongly encouraged to identify and acknowledge all forms of contribution to a project.

**Comment by Beanow** [comment]
> > should Contributors include code contributions
> 
> Yes, I think this is the way the term is commonly used (even in the rest of the sfosc text), so it would make sense to define it like this

I think so too. Trying to move towards Developers means to fight with GitHub/established terminology. But it shouldn't be so hard to expand on it instead.

> > Is Contributors the right term to be using for community members that don't contribute code?
> 
> This is perhaps less universal, but I think it is good to affirm non-code contributions are as important as (sometimes more important than) code contributions, so calling those who make non-code contributions "contributors" makes sense to me.

To some extent I agree. But I think there may be better terms in some situations. For example, you might have a business that is the only company that contributes paid development hours from their staff. It is a contribution, so the business can be called a contributor, but they may prefer to be called project sponsor or even claim "developed by".

> > Do we need more terms to cover this correctly
> 
> Not AFAICS

I think a few individual level terms are more natural than contributors, for example donors.
But mostly non-individuals may need some more. Partners, sponsors, collaborators, etc.
Splitting it up this far is starting to look more like what @lcooney suggested.

> I think ‘class’ of community is probably not a term we want to use. Maybe ‘Community Members’ with detailed titles?
> 
>   * Developers
>   * Docs WriterS
>   * Marketing / Events Managers
>   * etc

And I'm starting to like that structure more. Decide on a fitting catch-all term we can use in page headers and attribute fairly specific contribution types with an org or individual.

**Comment by Beanow** [comment]
> The term that complicates things is probably the Contributor one. Because it historically meant something else in this context and is _still reinforced_ by how tools like GitHub define and display it.
> 
> ![Contributor - This user has previously committed to the sfosc repository.](https://user-images.githubusercontent.com/497556/57445936-bf5e6f80-7253-11e9-89f7-244724210cf8.png)
> _(This badge is really unfortunate, we ought to lobby with GitHub about that.)_

@devonzuegel I saw [your talk](https://www.youtube.com/watch?v=n47rCa9dxf8) yesterday and thought you might be someone to ask about the above.

Do you think there's any room within GitHub to make some changes to these labels / badges?
I think they are very much stuck in the "code artifacts are what matters" mindset, but are a great way in terms of UI to do attribution. Perhaps we can reconfigure these to be more inclusive of other contribution types?

**Comment by devonzuegel** [comment]
I love this idea! Will share the idea with other folks at GitHub. Thanks for sharing and for reaching out, @Beanow.

**Comment by nothingismagick** [comment]
I think in the context of what Sourcecred is doing it might be useful to be able to DEFINE your own rules for contributions and contributor types - like how we can make our own Tags / define a security policy. This would probably sidestep the endless discussions necessary to tweak this from an operational point of view (aka GitHub).

If there's anything we can do to help out here, @devonzuegel - by all means let us know.
